### PR TITLE
Update skip check

### PR DESCRIPTION
**Pull request by daniel-butler-irl**
### Description

fixes bug where if a pr comes from a fork the commit messages were checked between the main/master of the fork and the pr. If the PR came from a fork on main/master there would be no commits listed. Now we add the upstream base repo, fetch the commits and check against the correct repo/braches

### Release required?

- [ ] No release
- [x] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

File: testhelper/tests.go [hunk 1]
```diff
@@ -25,11 +24,27 @@ import (
...
+	// Set upstream to the source repo
+	remote_out, remote_err := exec.Command("/bin/sh", "-c", "git remote add upstream "+source_repo).Output()
+	if remote_err != nil {
+		logger.Log(options.Testing, "Add remote output:\n", remote_out)
+		logger.Log(options.Testing, "Error adding upstream remote:\n", remote_err)
+		return false
+	}
+	// Fetch the source repo
+	fetch_out, fetch_err := exec.Command("/bin/sh", "-c", "git fetch upstream -f").Output()
+	if fetch_err != nil {
```
Comment (toddgiguere) [inline testhelper/tests.go:38]: Did you want to do anything here if the fetch fails?

File: testhelper/tests.go [hunk 2]
```diff
@@ -25,11 +24,27 @@ import (
 	"github.com/terraform-ibm-modules/ibmcloud-terratest-wrapper/common"
 )
 
-func skipUpgradeTest(branch string) bool {
+func (options *TestOptions) skipUpgradeTest(source_repo string, branch string) bool {
+
+	// Set upstream to the source repo
+	remote_out, remote_err := exec.Command("/bin/sh", "-c", "git remote add upstream "+source_repo).Output()
```
Comment (toddgiguere) [inline testhelper/tests.go:30]: Question: does this succeed if the fork already has a remote called "upstream"?

**Review by toddgiguere — CHANGES_REQUESTED** [review]

File: testhelper/tests.go (hunk 1 above)
Comment (daniel-butler-irl) [reply testhelper/tests.go:42]: I don't think so, I was just going to run the upgrade test if it cannot figure it out and log a message. I feel this is better than killing the test... But its open for discussion

**Review by toddgiguere — APPROVED** [review]

**Pull request merged**